lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -3933,6 +3933,19 @@ OpFoldResult AtenMulOp::fold(FoldAdaptor adaptor) {
       [](double a, double b) -> double { return a * b; });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenMulIntFloatOp
+//===----------------------------------------------------------------------===//
+
+OpFoldResult AtenMulIntFloatOp::fold(FoldAdaptor adaptor) {
+  if (!adaptor.getA() || !adaptor.getB()) {
+    return nullptr;
+  }
+  return atenBinaryFloatOperatorFoldHelper(
+      adaptor.getOperands(),
+      [](double a, double b) -> double { return a * b; });
```

Comment:
I used `atenBinaryFloatOperatorFoldHelper()` (which takes `std::function<double(double, double)>` as second argument) because it's also used by `AtenMulOp` (which multiplies two `Scalar`s) and by `AtenAddFloatIntOp`.
Note that `atenBinaryFloatOperatorFoldHelper()` is internally casting integer operands to double whenever necessary.

I avoided the simplification you mentioned because I saw that other folders, like `AtenMulOp`'s, don't do it. I'm not sure whether there is a specific reason. It might be worthwhile to tackle these simplification as a whole in a dedicated PR.

Reply:
@giacs-epic I think you are right.

Reply:
Ah, good to know